
Observation Simulator S/W Requirements

The Observation Simulator has some requirements in terms of software.
They are listed below:

Core Simulator
* MySQL 4.1 or greater
* Python 2.3 or greater with the following modules
    - MySQLdb (http://sourceforge.net/projects/mysql-python)
    - slalib (???)
    - SimPy (http://simpy.sourceforge.net/)
    - heapqc (http://zhar.net/projects/python/)
    - Sun (http://kortis.to/radix/python/code/Sun.py)

Plotting Utilities
* MySQL 4.1 or greater (http://www.mysql.com)
* Python 2.3 or greater (http://www.python.org) 
  with the following modules:
    - MySQLdb (http://sourceforge.net/projects/mysql-python)
    - slalib (???)
    - SimPy (http://simpy.sourceforge.net/)
    - heapqc (http://zhar.net/projects/python/)
    - Sun (http://kortis.to/radix/python/code/Sun.py)
    - Numeric (http://www.numpy.org/)
    - pyx (http://pyx.sourceforge.net/)
    - matplotlib (http://matplotlib.sourceforge.net/)
    - TableIO (http://php.iupui.edu/~mmiller3/python/)

Field List Generation Tools (FieldGen and AddGalEcl)
* SLALIB/C


Of these, the Observation Simulator distribution includes the following
Python modules
* Sun
* SimPy

For the other Python modules, compilation is necessary. For this reason
they are not included in the distribution. Instructions on how to 
compile and install them (as well as on their s/w requirements) are to
be found on their respective web sites.

===============================================================================
        Installation Gotcha's
===============================================================================
        PYREX  installation problem
===============================================================================
# Stuart comment regarding a problem with pyrex installation:
Special thanks go to google for finding:
   http://lists.copyleft.no/pipermail/pyrex/2005-February/001137.html
   which solved my 
   
   TypeError: swig_sources() takes exactly 2 arguments (3 given) 
problem.

	[Pyrex] bug in distutils, or a python bug?
	Charles Moad cmoad at indiana.edu
	Tue Feb 22 13:26:47 CET 2005
	
	    * Previous message: [Pyrex] bug in distutils, or a python bug?
	    * Messages sorted by: [ date ] [ thread ] [ subject ] [ author ]
	
	distutils was modified in 2.4.  This patch was posted on the list a 
	while back.
	
	--- build_ext.py        Fri Apr 30 08:00:46 2004
	+++ new-build_ext.py    Wed Dec 29 03:11:19 2004
	@@ -32,7 +32,7 @@
	        if self.compiler == 'mingw32':
	          self.swig_cpp = 1
	
	-  def swig_sources (self, sources):
	+  def swig_sources (self, sources, extension=None):
	      if not self.extensions:
	        return
	
	
	- Charlie
	
	Simon Wittber wrote:
	> I've just installed the free MS compiler using the instructions at:
	> vrplumber.com/programming/mstoolkit/
	> 
	> I've also installed:
	> http://www.prescod.net/pyximport/
	> 
	> Now for the weird error:
	>   File "c:\python\24\lib\distutils\command\build_ext.py", line 443, in
	> build_extension
	>     sources = self.swig_sources(sources, ext)
	> TypeError: swig_sources() takes exactly 2 arguments (3 given)
	> 
	> If you look closely at the above error, you will notice it is a method
	> call, and therefore, the self argument is passed implicitly.
	> 
	> The method signature contains 3 arguments, and looks like:
	> def swig_sources (self, sources, extension)
	> 
	> Any idea why this error is occuring? At forst glance, it looks like a
	> rather nasty python2.4 bug!
	> 
	> Sw.
===============================================================================
        matplotlib installation problem
===============================================================================
Stuart comments:
	Perhaps I have a too new version but I still have trouble with
	matplotlib.
	
	I changed:
	   from matplotlib.matlab import *
	to
	   from matplotlib.mlab import *

	since there was no matlab module/class in the matplotlab area.
	(I think they changed the name to avoid legal issues)
	and I had to add:

	   from pylab import *

	as shown on the matplotlab web page:
	   http://matplotlib.sourceforge.net/pylab_commands.html

	to get it to quit complaining about no "figure()" command
	defined.

This needs to be done in the following modules: 
    lsst/simulator/viz/{newextract.py newbri.py eqgal.py}
    lsst/simulator/src/{extractdata.py plotfields.py}
    
